R-package/tests/testthat/test_dataset.R

```diff
@@ -205,3 +205,51 @@ test_that("Dataset$update_params() works correctly for recognized Dataset parame
     expect_identical(new_params[[param_name]], updated_params[[param_name]])
   }
 })
+
+test_that("lgb.Dataset: should be able to run lgb.train() immediately after using lgb.Dataset() on a file", {
+  dtest <- lgb.Dataset(data = test_data
+                       , label = test_label)
+  tmp_file <- tempfile(pattern = "lgb.Dataset_")
+  lgb.Dataset.save(dataset = dtest
+                   , fname = tmp_file)
+  
+  # read from a local file
+  dtest_read_in <- lgb.Dataset(data = tmp_file)
+  
+  param <- list(
+    objective = "binary"
+    , metric = "binary_logloss"
+    , num_leaves = 5L
+    , learning_rate = 1.0
+  )
+  
+  # should be able to train right away
+  bst <- lgb.train(params = param
+                   , data = dtest_read_in)
+  
+  expect_true(lgb.is.Booster(x = bst))
+})
+
+test_that("lgb.Dataset: should be able to run lgb.cv() immediately after using lgb.Dataset() on a file", {
+  dtest <- lgb.Dataset(data = test_data
+                       , label = test_label)
+  tmp_file <- tempfile(pattern = "lgb.Dataset_")
+  lgb.Dataset.save(dataset = dtest
+                   , fname = tmp_file)
+  
+  # read from a local file
+  dtest_read_in <- lgb.Dataset(data = tmp_file)
+  
+  param <- list(
+    objective = "binary"
+    , metric = "binary_logloss"
+    , num_leaves = 5L
+    , learning_rate = 1.0
+  )
+  
+  # should be able to train right away
+  bst <- lgb.cv(params = param
+                , data = dtest_read_in)
+  
+  expect_true(lgb.is.Booster(x = bst))
```

Comment:
Hi @jameslamb, it seems to be failing the tests with this change. I re ran the test locally and got the error:
```
Error: `bst` inherits from `lgb.CVBooster/R6` not `CVBooster`.
```

If I update the code to 
`expect_is(bst, "lgb.CVBooster")`

it passes. Are you ok with that?

Reply:
yep, that's correct and it's what we use in other tests: https://github.com/microsoft/LightGBM/blob/c5d9d2436b92a686f76c1d77de2d22e766e0e599/R-package/tests/testthat/test_learning_to_rank.R#L92

Sorry I misled you! That's what I get for trying to type it out from memory instead of just testing locally myself 😬

Reply:
No problem at all, should be good to review now I think!